pkg/registries/registries.go

```diff
@@ -96,26 +105,47 @@ func mergedMirrorSets(icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy
 }
 
 // EditRegistriesConfig edits, IN PLACE, the /etc/containers/registries.conf configuration provided in config, to:
-// - Mark whole registries in insecureRegistries as insecure (TLS is not required, and TLS certificate verification is not required when TLS is used)
-// - Mark whole registries in blockedRegistries as blocked (any attempts to access them fail)
+// - Mark whole registries or wildcard entries in insecureRegistries as insecure (TLS is not required, and TLS certificate verification is not required when TLS is used)
+// - Mark whole registries or wildcard entries in blockedRegistries as blocked (any attempts to access them fail)
 // - Implement ImageContentSourcePolicy rules in icspRules.
 // “Whole registries” above means that the configuration applies to everything on that registry, including any possible separately-configured
 // namespaces/repositories within that registry.
+// "Wildcard entries" above means that we accept wildcards in the form of *.example.registry.com for insecure, blocked, and allowed registries only. We do not
+// accept them for mirror configuration.
 func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegistries, blockedRegistries []string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) error {
+
+	// addRegistryEntry creates a Registry object corresponding to scope.
+	addRegistryEntry := func(scope string) {
+		// If scope is a wildcard entry, add it to the registry Prefix
+		if strings.HasPrefix(scope, "*.") {
+			config.Registries = append(config.Registries, sysregistriesv2.Registry{
+				Prefix: scope,
+			})
+			// Otherwise it is a regular entry so add it to the registry endpoint Location
+		} else {
+			config.Registries = append(config.Registries, sysregistriesv2.Registry{
+				Endpoint: sysregistriesv2.Endpoint{Location: scope},
+			})
+		}
+	}
+
 	// getRegistryEntry returns a pointer to a modifiable Registry object corresponding to scope,
 	// creating it if necessary.
-	// NOTE: We never generate entries with Prefix != Location, so everything in updateRegistriesConfig
-	// only checks Location.
+	// Note: Prefix can be a wildcard entry in the format "*.example.registry.com" so we check the value of
+	// Prefix for a wildcard entry. If Prefix doesn't have a wildcard entry, we check Location for regular entries.
 	// NOTE: The pointer is valid only until the next getRegistryEntry call.
+	// NOTE: Validation of wildcard entries is done before EditRegistriesConfig is called in the MCO code.
```

Comment:
(Absolutely non-blocking: This applies to all of `EditRegistriesConfig`, so it might be better placed in the documentation of the whole function, or just inside the function body if it is intended to document the implementation approach and not the API.)

Reply:
Fixed